service/routers/users.js

```diff
@@ -27,34 +27,32 @@ const getUsers = async (req, res) => {
 }
 
 const getUsersById = async (req, res) => {
-  const requestor = await User.findById(userId)
   const { userId } = req.user
-  if (requestor.role === 'admin' || requestor.role === 'superuser') {
-    const user = await User.findById(req.params.id)
-  } else if(requestor.role === 'user' && requestor.id === user.id){
-    res.send(sanitizerUsers(users))
-  } else{}
-  try {
-    const user = await User.findById(req.params.id)
-    res.send(sanitizerUsers([user]))
-  } catch (err) {
-    console.log(`Error getting usiser by id: ${err}`)
-    res.sendStatus(500)
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || req.user.role === 'superUser' || requestor._id.toString() === req.params.id.toString()) {
+    const users = await User.findById(req.params.id)
+    res.send(users)
+  } else {
+    res.status(404).send('This route is only for admin and superuser')
```

Comment:
The status code for a user trying to access a resource they're not supposed to is 403 forbidden. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403